version/version.go

```diff
@@ -10,7 +10,7 @@ const (
 
 	// TMVersionDefault is the used as the fallback version of Tendermint Core
 	// when not using git describe. It is formatted with semantic versioning.
-	TMVersionDefault = "0.34.11"
+	TMVersionDefault = "0.35.0-unreleased"
```

Comment:
this is handled in the makefile (https://github.com/tendermint/tendermint/blob/master/Makefile#L9). No need to change here

Reply:
Can you explain a little more what the makefile does? This is just for the `tendermint version` command right? What happens if users import this value directly for their applications?

Reply:
This is only the default if not set via ldflags. If a user is using master or an unreleased version with ldflags then this will be overwritten with anything they set. The default is merely meant as a way to not require apps to add the ldflag since many people were forgetting. 

It is fine to add this, but since we dont recommend using master or any unreleased version, bad practice anyways, if a team does this they are responsible for any instability

Reply:
Thanks for the explanation. Yeah I thought with this change that at least if someone is using it they don't mistake master for 0.34.11

Reply:
Agreed—it makes sense to have a non-confusing default for `go build`.